apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php

```diff
@@ -467,13 +467,37 @@ public function merge_template_and_post( $post ) {
 
 		$template = new A8C_WP_Template( $post->ID );
 
-		// Bail if the post has no tempalte id assigned.
+		// Bail if the post has no template id assigned.
 		if ( ! $template->get_template_id() ) {
 			return;
 		}
 
-		$wrapped_post_content = sprintf( '<!-- wp:a8c/post-content -->%s<!-- /wp:a8c/post-content -->', $post->post_content );
-		$post->post_content   = str_replace( '<!-- wp:a8c/post-content {"align":"full"} /-->', $wrapped_post_content, $template->get_template_content() );
+		$parser           = new WP_Block_Parser();
+		$template_content = $template->get_template_content();
+		$template_blocks  = $parser->parse( $template_content );
```

Comment:
If this is expensive it looks like we can check before parsing if we have a related block.

`has_blocks` https://core.trac.wordpress.org/browser/tags/5.2/src/wp-includes/blocks.php#L81 (this does a simple string pos check)

Reply:
🤔 should we be calling parse_blocks https://core.trac.wordpress.org/browser/tags/5.2/src/wp-includes/blocks.php#L286 

The main difference is that this allows a hookpoint to change the parser implementation.